src/lightning/pytorch/CHANGELOG.md

```diff
@@ -40,6 +40,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 ### Fixed
 
 - Fixed issue where the `precision="transformer-engine"` argument would not replace layers by default ([#19082](https://github.com/Lightning-AI/lightning/pull/19082))
+- Fixed issue where layers created in `LightningModule.setup` or `LightningModule.configure_model` wouldn't get converted when using the Bitsandbytes or TransformerEngine plugins ([#19061](https://github.com/Lightning-AI/lightning/pull/19061))
```

Comment:
I think it's probably safer if we document under "Changed" that the model gets moved to the device now before optimizers get setup (I believe only single-device and deepspeed is affected by this?). I'm not sure this breaks anything.

Reply:
I would not consider this worth mentioning as it's not something we advertised before (or now) and we don't know of any issues related to the change.

It's basically an internal implementation detail of each strategy

Reply:
Just mentioning this because there are strategies out there that override our DDP: https://github.com/NVIDIA/NeMo/blob/d3237d557b752a5eb1631f1cee10fd26128c3933/nemo/collections/nlp/parts/nlp_overrides.py#L147
If you notice there, it's calling super().